Sources/OpenPass/IDTokenValidation.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  IDTokenValidation.swift
+//
+// MIT License
+//
+// Copyright (c) 2024 The Trade Desk (https://www.thetradedesk.com/)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+import Foundation
+
+/// Validates an IDToken against JWKS.
+protocol IDTokenValidation {
+    func validate(_ token: IDToken, jwks: JWKS) throws -> Bool
+}
+
+/// Validates the following:
+/// - Token is not expired
+/// - Token issue date is not in the past
+/// - Token signature is correct according to JWKS
+///  https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
+internal struct IDTokenValidator: IDTokenValidation {
+    /// Set a specific leeway window in seconds in which the Expires At ("exp") Claim will still be valid.
+    private let expiresAtLeeway: Double = 0
+
+    /// Set a specific leeway window in seconds in which the Issued At ("iat") Claim will still be valid. This method
+    /// overrides the value set with acceptLeeway(long). By default, the Issued At claim is always verified
+    /// when the value is present
+    private let issuedAtLeeway: Double = 60
+
+    private let dateGenerator: DateGenerator
+
+    init(dateGenerator: DateGenerator = .init { Date() }) {
+        self.dateGenerator = dateGenerator
+    }
+
+    func validate(_ token: IDToken, jwks: JWKS) throws -> Bool {
```

Comment:
Just comparing to Android's `IDTokenVerifier`, I think we're missing at least an audience (AUD) check against the client id? and potentially the token "issuer"?